src/jdmtool/featunlk.py

```diff
@@ -402,31 +412,70 @@ def display_content_of_dat_file(dat_file: pathlib.Path):
+            print('** ' + footer_bytes[0x20:0x20+11].decode('ascii'))
+            print('** ' + footer_bytes[0x2B:0x2B+20].decode('ascii'))
+            print('** ' + footer_bytes[0x98:0x98+20].decode('ascii'))
+    elif feature in (Feature.TERRAIN,):
+        print(f"DB_MAGIC: 0x{int.from_bytes(header_bytes[0:4], 'little'):08X}")
+        print('** ' + header_bytes[0x58:0x58+20].decode('ascii'))
+        print('** Cycle: ' + footer2_bytes[0x1:0x1+4].decode('ascii'))
+        print('** ' + header_bytes[0x78:0x78+12].decode('ascii'))
+        print('** ' + header_bytes[0x86:0x86+4].decode('ascii'))
+        print('** ' + header_bytes[0x8c:0x8c+4].decode('ascii'))
+    elif feature in (Feature.OBSTACLE2, Feature.SAFETAXI2):
         if DB_MAGIC != int.from_bytes(footer_bytes[0:4], 'little'):
             print('WRONG MAGIC!!')
             print(f"0x{int.from_bytes(footer_bytes[0:4], 'little'):08X}")
-        print('** ' + footer_bytes[-0x6a:-0x61].decode('ascii') + ' ' +
-              footer_bytes[4:8].decode('ascii') + ' ' + '\n** ' + footer_bytes[28:43].decode('ascii') +
-              '\n** ' + footer_bytes[43:55].decode('ascii'))
+        print('** ' + footer_bytes[-0x6a:-0x61].decode('ascii')) 
+        print('** ' + footer_bytes[4:8].decode('ascii'))
+        print('** ' + footer_bytes[28:43].decode('ascii'))
+        print('** ' + footer_bytes[43:43+30].decode('ascii'))
+        print('** ' + footer_bytes[152:152+20].decode('ascii'))
         (f_month, f_day, f_year) = struct.unpack('<BBH', footer_bytes[-0xFA:-0xFA+0x4])
         (t_month, t_day, t_year) = struct.unpack('<BBH', footer_bytes[-0xF6:-0xF6+0x4])
-        print(f'** Effective: {f_year}-{f_month:02}-{f_day:02} to {t_year}-{t_month:02}-{t_day:02}')
+        cus_date1 = (datetime.strptime(f'{f_day:02}{f_month:02}{f_year}', "%d%m%Y").date()).strftime("%d-%b-%Y")
+        cus_date2 = (datetime.strptime(f'{t_day:02}{t_month:02}{t_year}', "%d%m%Y").date()).strftime("%d-%b-%Y")
+        print(f'** Effective {cus_date1} to {cus_date2}')
+        
+        
     elif feature in (Feature.AIRPORT_DIR, ):
-        if DB_MAGIC2 != int.from_bytes(footer_bytes[0:4], 'little'):
+        if (DB_MAGIC2 == int.from_bytes(footer_bytes[0:4], 'little')):
+            print('** ' + header_bytes[0x54:0x54+40].decode('ascii'))
+            timestamp = int.from_bytes(header_bytes[0x10:0x10+4], 'little')
+        if (str(dat_file) == 'apt_dir.gca'): 
+            print('** Cycle: ' + footer_bytes[0x4:0x4+4].decode('ascii'))
+            print('** ' + footer_bytes[0x20:0x20+11].decode('ascii'))
+            print('** ' + footer_bytes[0x2B:0x2B+20].decode('ascii'))
+            print('** ' + footer_bytes[0x98:0x98+20].decode('ascii'))
+        if (DB_MAGIC2 != int.from_bytes(footer_bytes[0:4], 'little') and 
+            (str(dat_file) != 'apt_dir.gca')):
             print('WRONG MAGIC!!')
             print(f"0x{int.from_bytes(footer_bytes[0:4], 'little'):08X}")
```

Comment:
At least in my test case, the `apt_dir.gca` has the magic bytes equal to `DB_MAGIC`. Does this logic work in your case? If so, that's better than checking the filename:
```
        if DB_MAGIC2 == int.from_bytes(footer_bytes[0:4], 'little'):
             ...
        elif DB_MAGIC == int.from_bytes(footer_bytes[0:4], 'little'):
             ...
        else:
             ...
```